content/en/blog/_posts/2024-03-15-userns-beta/index.md

```diff
@@ -0,0 +1,135 @@
+are moving to beta!
+
+## What is a user namespace?
+
+A user namespace is a Linux feature that isolates the user and group identifiers
+(UIDs and GIDs) of the containers from the ones on the host. The identifiers in
+the container can be mapped to identifiers on the host in a way where the host
+UID/GIDs used for different containers never overlap. Furthermore, the
+identifiers can be mapped to unprivileged non-overlapping UIDs and GIDs on the
+host. This brings two key benefits:
+
+ * _Prevention of lateral movement_: As the UIDs and GIDs for different
+containers are mapped to different UIDs and GIDs on the host, containers have a
+harder time attacking each other even if they escape the container boundaries.
+For example, if container A is running with different UIDs and GIDs on the host
+than container B, the operations it can do on container B’s files and process
+are limited: only read/write what a file allows to others, as it will never
+have permission owner or group permission (the UIDs/GIDs on the host are
+guaranteed to be different for different containers).
+
+ * _Increased host isolation_: As the UIDs and GIDs are mapped to unprivileged
+users on the host, if a container escapes the container boundaries, even if it
+is running as root inside the container, it has no privileges on the host. This
+greatly protects what host files it can read/write, which process it can send
+signals to, etc. Furthermore, capabilities granted are only valid inside the
+user namespace and not on the host, which also limits the impact a container
+escape can have.
+
+![UIDs/GIDs layout for userns](userns-ids.jpg)
+
+
+Without using a user namespace, a container running as root in the case of a
+container breakout, has root privileges on the node. And if some capabilities
+were granted to the container, the capabilities are valid on the host too. None
+of this is true when using user namespaces (modulo bugs, of course 🙂).
+
+## Changes in 1.30
+
+The only change in Kubernetes 1.30, besides moving to Beta, is that we now allow
+the kubelet to use custom ranges for the UIDs/GIDs mapping.
+
+See [the documentation][userns-doc] for how to configure custom ranges to use
+for the mapping.
+
+[userns-doc]: https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#set-up-a-node-to-support-user-namespaces
+
+## Demo
+
+A few weeks ago, [CVE-2024-21626][runc-cve] was disclosed. This **vulnerability
+score is 8.6 (HIGH)** and allows an attacker to escape a container and
+**read/write to any path on the node and other pods hosted on the same node**.
+
+Rodrigo created a demo which exploits [CVE 2024-21626][runc-cve] and shows how
+the exploit that works without user namespaces, **it is mitigated when user
+namespaces are in use.**
+
+{{< youtube id="" title="Mitigation of CVE-2024-21626 on Kubernetes by enabling User Namespace">}}
+
+Please note that with user namespaces we can do on the host file-system what the
+permission bits for "others" allows us.
```

Comment:
(nit)
Only use “we” to mean “the blog authors” or “the entire Kubernetes project”; don't include the reader in “we”:
```suggestion
Please note that with user namespaces you can do on the host file-system what the
permission bits for "others" allows you to do.
```

If you did mean the blog authors, I got it wrong and please skip this feedback.

Reply:
Right. An attacker seems more clear, updated to:

```
Please note that with user namespaces an attacker can do on the host file-system
what the permission bits for "others" allows.
```